mlflow/models/dependencies_schemas.py

```diff
@@ -55,23 +54,31 @@ def set_retriever_schema(
     retriever_schema = globals().get(DependenciesSchemasType.RETRIEVERS.value, [])
 
     # Check if a retriever schema with the same name already exists
-    if any(schema["name"] == name for schema in retriever_schema):
-        # reset if there is an error to clear the global state for next run
-        _clear_retriever_schema()
-        raise MlflowException(
-            f"A retriever schema with the name '{name}' already exists.",
-            error_code=ALREADY_EXISTS,
+    existing_schema = next((schema for schema in retriever_schema if schema["name"] == name), None)
```

Comment:
```suggestion
    existing_schema = next((schema for schema in retriever_schemas if schema["name"] == name), None)
```

can we use the plural form?